pkg/replication/health_aggregator.go

```diff
@@ -79,6 +80,18 @@ func (p *podHealth) GetHealth(host types.NodeName) (health.Result, bool) {
 	return h, ok
 }
 
+func (p *podHealth) NumHealthy() int {
```

Comment:
I can't figure out where this scopes its node list to the nodes targeted by the daemon set. I predict that you'll find that this overcounts the number of healthy hosts.